src/stream/src/executor/backfill/cdc/cdc_backfill.rs

```diff
@@ -269,11 +259,29 @@ impl<S: StateStore> CdcBackfillExecutor<S> {
             // otherwise the upstream changelog may be blocked by the snapshot read stream
             let _ = Pin::new(&mut upstream).peek().await;
 
-            tracing::info!(
-                upstream_table_id,
-                initial_binlog_offset = ?last_binlog_offset,
-                "start the bacfill loop");
+            // wait for a barrier to make sure the backfill starts after upstream source
```

Comment:
How do we know that upstream source will start after a barrier?

Reply:
strictly speaking, wait for a barrier doesn't imply that the source connector has subscribed to binlog if the barrier is too short. Ideally, we should wait for the initial cdc heartbeat message. Let me rethink whether it is feasible.